frontend/src/app/reducers/localizations.js

```diff
@@ -1,34 +1,60 @@
 // @flow
 
+// TODO: This Action type doesn't work below in the reducer
+// import type { Action } from 'testpilot/types';
+
 export type Localizations = {
   [language: string]: string
 };
 
-function defaultState(): Localizations {
-  return {
-  };
-}
+export type NegotiatedLanguages = Array<string>;
+
+export type LocalizationState = {
+  localizations: Localizations,
+  negotiatedLanguages: NegotiatedLanguages
+};
 
 export type SetLocalizationsAction = {
   type: 'SET_LOCALIZATIONS',
   payload: Localizations
 };
 
+export type SetNegotiatedLanguagesAction = {
+  type: 'SET_NEGOTIATED_LANGUAGES',
+  payload: NegotiatedLanguages
+};
+
+function defaultState(): LocalizationState {
+  return {
+    localizations: {},
+    negotiatedLanguages: []
+  };
+}
+
 function setLocalizations(
-  state: Localizations,
+  state: LocalizationState,
   action: SetLocalizationsAction
-): Localizations {
-  return action.payload;
+): LocalizationState {
+  return { ...state, localizations: action.payload };
+}
+
+function setNegotiatedLanguages(
+  state: LocalizationState,
+  action: SetNegotiatedLanguagesAction
+): LocalizationState {
+  return { ...state, negotiatedLanguages: action.payload };
 }
 
 export default function localizationsReducer(
-  state: ?Localizations,
-  action: SetLocalizationsAction
-): Localizations {
+  state: ?LocalizationState,
+  action: any // TODO: figure out how to correctly assert any Action here
```

Comment:
The longer explanation, for completeness' sake:

I noticed that flow-type-annotated reducers in our code assert this action should be the set of actions notionally associated with the reducer.

The problem, generally speaking, is that *any* action is valid here. *Any* reducer can handle *any* action - and in fact *all* reducers see *all* actions passing through `dispatch()`. It's the switch statement that dynamically decides when to do something. This doesn't fail flow type linting, because the chain of code which makes that happen is buried inside vendor code in Redux.

But, when I try to assert the more generic `Action` type here, I don't know how to type-case it into the specific action types for the other sub-functions. I get errors when I try.